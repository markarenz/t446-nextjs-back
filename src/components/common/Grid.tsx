import styles from '../../styles/modules/Grid.module.scss';

type Props = {
  columns: number;
  children: JSX.Element[];
};

const Grid: React.FC<Props> = ({ children, columns }) => {
  return (
    <div className={styles.root} data-testid="grid-wrap">
      {children}
    </div>
  );
};

export default Grid;
