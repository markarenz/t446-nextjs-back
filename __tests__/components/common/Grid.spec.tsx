import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import Grid from '../../../src/components/common/Grid';

const defaultProps = {
  columns: 3
};

describe('Grid', () => {
  it('renders grid wrapper', async () => {
    render(
      <Grid {...defaultProps}>
        <div>A</div>
        <div>B</div>
        <div>C</div>
      </Grid>
    );
    const element = await screen.findByTestId('grid-wrap');
    expect(element).toBeInTheDocument();
  });
});
