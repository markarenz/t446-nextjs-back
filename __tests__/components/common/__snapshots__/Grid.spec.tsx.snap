// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders grid wrapper 1`] = `
<div
  class="root"
  data-testid="grid-wrap"
>
  <div>
    A
  </div>
  <div>
    B
  </div>
  <div>
    C
  </div>
</div>
`;
